lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResultImpl.cs

```diff
@@ -120,29 +150,28 @@ internal static string GetTrackingUrl(string filePath)
             return result.Item2;
         }
 
-        internal string GetDriverUrlForYarn(String filePath)
+        internal string GetDriverUrlForYarn(string filePath)
         {
-            _driverUrl = GetTrackingUrl(filePath);
-            return _driverUrl;
+            return GetTrackingUrlAppId(filePath);
         }
 
         internal string GetDriverUrlForLocalRuntime(string filePath)
         {
-            _driverUrl = null;
+            string fullDriverUrl = null;
             for (int i = 0; i < 10; i++)
             {
                 var driverUrl = TryReadHttpServerIpAndPortFromFile(filePath);
                 if (!string.IsNullOrEmpty(driverUrl))
                 {
-                    _driverUrl = "http://" + driverUrl + "/";
+                    fullDriverUrl = "http://" + driverUrl + "/";
                     break;
                 }
                 Thread.Sleep(1000);
```

Comment:
Why the sleep? I'm very hesitant to add real time units anywhere to the code, as they will create bugs in the future.